networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1139 @@
+                        np.max(A),
+                        (np.max(A - np.min(A))) / num_thresholds,
+                    ),
+                    4,
+                )
+            )  # linear distribution
+        else:
+            thresholds = _distribute(
+                dist=threshold_distribution,
+                end_value_range=(np.min(A), np.max(A)),
+                n=num_thresholds,
+            )
+        # Converting to binary nx_graphs according to thresholds:
+        nx_graphs = [
+            nx.from_numpy_matrix(np.where(A > threshold, 1, 0), create_using=nx.DiGraph)
+            for threshold in thresholds
+        ]
+    # removes isolated nodes (0 in & out degree) from binary nodes. (not needed for consolidation)
+    for G in nx_graphs:
+        G.remove_nodes_from(list(nx.isolates(G)))
+
+    # eliminates empty graphs
+    nx_graphs = [graph for graph in nx_graphs if not nx.is_empty(graph)]
+    condensed_graphs = [nx.condensation(G) for G in nx_graphs]
+    largest_condensed_graphs = []
+    for condensed_graph in condensed_graphs:
+        largest_condensed_graphs.append(
+            nx.convert_node_labels_to_integers(
+                condensed_graph.subgraph(
+                    max(nx.weakly_connected_components(condensed_graph), key=len)
+                )
+            ).copy()
+        )
+        for node, attrs in largest_condensed_graphs[-1].nodes.data():
+            attrs["weight"] = len(attrs["members"])
+
+    return largest_condensed_graphs, nx_graphs
+
+
+@not_implemented_for("undirected")
+def weight_nodes_by_condensation(condensed_graph):
+    """Weights nodes according to the number of other nodes they condensed (sum of constituent cycle of DAG node).
+
+    As, proposed in _[1]:  e.g. if a cycle contained 3 nodes (and became one in condensation)
+    the resulting node of the condensed graph would then gain weight = 3.
+    Single (non-cyclic) nodes are weighted as 1.
+
+    Parameters
+    ----------
+    condensed_graph: NetworkX Graph
+        result of a networkx.condensation call, such that the 'members'
+        attribute is populated with constituent cyclical nodes
+
+    Return
+    ------
+    condensed_graph: NetworkX Graph
+        node weighted condensed graph
+
+    Examples
+    --------
+    Visualization also recommended here (with node size prop. weighting)
+    >>> import numpy as np
+    >>> b = np.array([
+    ...     [0, 0.2, 0, 0, 0],
+    ...     [0, 0, 0, 0.7, 0],
+    ...     [0, 0.4, 0, 0, 0],
+    ...     [0, 0, 0.1, 0, 1.0],
+    ...     [0, 0, 0, 0, 0],
+    ... ])
+
+    >>> num_thresholds = 2
+    >>> condensed_networks, _ = nx.node_weighted_condense(b, num_thresholds=num_thresholds)
+    >>> for network_index in range(num_thresholds):
+    ...     for node_index in range(len(condensed_networks[network_index].nodes)):
+    ...         print(f"Node {node_index}, new weight:", condensed_networks[network_index].nodes[node_index]["weight"])
```

Comment:
```suggestion
    >>> for cG in condensed_networks:
    ...     for node_index, (node, weight) in enumerate(cG.nodes.data("weight")):
    ...         print(f"Node {node}, new weight: {weight}")
```
I couldn't easily see whether the node_index is needed here. If condensed_networks has integer nodes then it isn't.

Reply:
You're right; no need for the `node_index`; implemented as suggested.

Reply:
You're right; no need for the `node_index`; implemented as suggested.